packages/babel-core/src/transform-file.js

```diff
@@ -25,15 +25,7 @@ type TransformFile = {
 
 const transformFileRunner = gensync<[string, ?InputOptions], FileResult | null>(
   function*(filename, opts) {
-    let options;
-    if (opts == null) {
-      options = { filename };
-    } else if (opts && typeof opts === "object") {
-      options = {
-        ...opts,
-        filename,
-      };
-    }
+    const options = { ...opts, filename };
```

Comment:
This isn't really related, but since `opts` can only be `null` or an object these two things have the same behavior.